src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -91,6 +92,39 @@ def _inner_not_found(ex):
                                  doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
                                  supports_no_wait=True)
 
+    with self.command_group('cdn endpoint rule', cdn_endpoints_sdk) as g:
+        g.show_command('show', 'get')
+        g.generic_update_command('add', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='add_rule',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
+        g.generic_update_command('remove', setter_name='update', setter_arg_name='endpoint_update_properties',
+                                 custom_func_name='remove_rule',
+                                 doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
+                                 supports_no_wait=True)
```

Comment:
why do you register these commands using generic_update_command?

Reply:
Because these are actually update endpoint calls which is a long running operation.